System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryTests.cs

```diff
@@ -1371,6 +1371,31 @@ public void Move_DirectoryExistsWithDifferentCase_DirectorySuccessfullyMoved()
             Assert.IsTrue(fileSystem.File.Exists(XFS.Path(@"C:\NewLocation\Data\someFile.txt")));
         }
 
+        public enum FileSystemType
+        {
+            RealFileSystem,
+            MockFileSystem
+        }
+
+        [TestCase(FileSystemType.RealFileSystem)]
+        [TestCase(FileSystemType.MockFileSystem)]
+        public void Move_Works(FileSystemType fileSystemType)
+        {
+            var tempDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            var sourceDir = Path.Combine(tempDir, "Foo", "Bar");
+            var targetDir = Path.Combine(tempDir, "Foo", "Baz");
+
+            var fileSystem = fileSystemType == FileSystemType.RealFileSystem
+                ? (IFileSystem)new FileSystem()
+                : new MockFileSystem();
+
+            fileSystem.Directory.CreateDirectory(sourceDir);
+
+            Assert.DoesNotThrow(() =>
+                fileSystem.Directory.Move(sourceDir, targetDir)
+            );
```

Comment:
@pianomanjh This assertion fails when using the mock FS and passes when using the real FS on both Windows & Linux (see the PR build).

Did you test differently?

Reply:
I'm taking another look now to see what's up, I've added a matrix of source/target paths now to cover more combinations...

Reply:
yeah, I can confirm the regression.  I can add some commits here with the fix and much more coverage around the characteristics of the source folder.  It's quite tricky sadly.  Copying from `c:\foo\bar` to `c:\foo\baz` does work, but from `c:\foo\bar` to `c:\foo2\baz` does not.

Reply:
there is also different behavior based on whether or not your user principal created 'all' the folders in the source path, which is how my testing was flawed.  I won't address that situation with this work, and make the assumption that all folders are permissioned.

Reply:
I posted the fix here: https://github.com/System-IO-Abstractions/System.IO.Abstractions/pull/580